extension/encoding/awscloudwatchmetricstreamsencodingextension/extension_test.go

```diff
@@ -16,17 +16,23 @@ func TestNew_JSON(t *testing.T) {
 	require.NoError(t, err)
 	require.NotNil(t, e)
 
-	_, err = e.UnmarshalMetrics([]byte{})
-	assert.EqualError(t, err, `failed to unmarshal metrics as 'json' format: 0 metrics were extracted from the record`)
+	expectedErrStr := "failed to unmarshal metrics as 'json' format: " +
+		"error unmarshaling datum at index 0: readObjectStart: " +
+		"expect { or n, but found a, error found in #1 byte of ...|a|..., " +
+		"bigger context ...|a|..."
```

Comment:
If there's something slightly less implementation-specific we could match using `assert.ErrorContains`, that would be great.